corehq/apps/userreports/expressions/getters.py

```diff
@@ -75,11 +76,31 @@ def recursive_lookup(dict_object, keys):
 
 def transform_date(item):
     # postgres crashes on empty strings, but is happy to take null dates
-    return item or None
+    if item:
+        if isinstance(item, basestring):
+            try:
+                return datetime.strptime(item, '%Y-%m-%d').date()
+            except ValueError:
+                pass
+        elif isinstance(item, date):
+            return item
+    return None
 
 
 def transform_datetime(item):
-    return item or None
+    if item:
+        if isinstance(item, basestring):
+            for datetime_format in [
+                '%Y-%m-%dT%H:%M:%SZ',
```

Comment:
@dannyroberts I followed the example of ```DateTimeProperty``` - do these formats correspond to those in use in HQ?